echallan-services/src/main/java/org/egov/echallan/repository/ServiceRequestRepository.java

```diff
@@ -0,0 +1,42 @@
+package org.egov.echallan.repository;
+
+import java.util.Map;
+
+import org.egov.tracer.model.ServiceCallException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Repository
+@Slf4j
+public class ServiceRequestRepository {
+
+	@Autowired
+	private RestTemplate restTemplate;
+	
+	@Autowired
+	private ObjectMapper mapper;
+	
+	public Object fetchResult(StringBuilder uri, Object request) {
+		mapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+		Object response = null;
+		log.info("URI: "+uri.toString());
+		try {
+			log.info("Request: "+mapper.writeValueAsString(request));
+			response = restTemplate.postForObject(uri.toString(), request, Map.class);
+		}catch(HttpClientErrorException e) {
```

Comment:
one catch block is throwing error, other one doesn't. i hope the null response from this method will be handled by the caller.

Reply:
Changes commited